plugins/huaweicloud/src/main/java/org/eclipse/xpanse/orchestrator/plugin/huaweicloud/HuaweiTerraformResourceHandler.java

```diff
@@ -0,0 +1,93 @@
+            var stateFile = deployResult.getRawResources().get("stateFile");
+            if (!stateFile.getClass().equals(String.class)) {
+                throw new RuntimeException("stateFile is unsupported.");
+            }
+            tfState = objectMapper.readValue(stateFile.toString(), TfState.class);
+        } catch (IOException ex) {
+            log.error("Parse terraform state content failed.");
+            throw new TerraformExecutorException("Parse terraform state content failed.", ex);
+        }
+        for (TfStateResource tfStateResource : tfState.getResources()) {
+            DeployResourceEntity deployResourceEntity = new DeployResourceEntity();
+            if (tfStateResource.getType().equals("huaweicloud_compute_instance")) {
+                for (TfStateResourceInstance instance : tfStateResource.getInstances()) {
+                    deployResourceEntity.setKind(DeployResourceKind.Vm);
+                    deployResourceEntity.setId((String) instance.getAttributes().get("id"));
+                    deployResourceEntity.setName((String) instance.getAttributes().get("name"));
+
+                    deployResourceEntity.setProperty(new HashMap<>());
+                    deployResourceEntity.getProperty()
+                            .put("ipv4", (String) instance.getAttributes().get("access_ip_v4"));
+                    deployResourceEntity.getProperty()
+                            .put("image_id", (String) instance.getAttributes().get("image_id"));
+                    deployResourceEntity.getProperty()
+                            .put("image_name", (String) instance.getAttributes().get("image_name"));
+                    deployResourceEntity.getProperty()
+                            .put("region", (String) instance.getAttributes().get("region"));
+                }
+            }
+            if (tfStateResource.getType().equals("huaweicloud_compute_eip_associate")) {
+                for (TfStateResourceInstance instance : tfStateResource.getInstances()) {
+                    deployResourceEntity.setProperty(new HashMap<>());
+                    deployResourceEntity.getProperty()
+                            .put("ip", (String) instance.getAttributes().get("public_ip"));
+                    deployResourceEntity.setKind(DeployResourceKind.PublicIp);
+                }
+            }
+            if (tfStateResource.getType().equals("huaweicloud_vpc")) {
+                for (TfStateResourceInstance instance : tfStateResource.getInstances()) {
+                    DeployResourceEntity xpResource = new DeployResourceEntity();
+                    xpResource.setKind(DeployResourceKind.Vpc);
+                    xpResource.setId((String) instance.getAttributes().get("id"));
+                    xpResource.setName((String) instance.getAttributes().get("name"));
+                }
+            }
+            xpResourceList.add(deployResourceEntity);
+        }
+
+        deployResult.setResources(xpResourceList);
```

Comment:
the entities which we created are not saved to the DB here?

Reply:
yes, not here. the entities should be saved in the skeleton.